### PR TITLE
Integrate coinbase alternative - pulled out invalidateCallback as util

### PR DESCRIPTION
**Pull request by CBarrett2**
Sorry for the extra PR, but wasn't sure if this was what you meant @tlsim :)

### CODE REVIEW

File: src/assets/js/data/feed/coinbase/historicFeed.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+(function(sc) {
+    'use strict';
+
+    sc.data.feed.coinbase.invalidator = function() {
```
Comment (CBarrett2) [inline src/assets/js/data/feed/coinbase/historicFeed.js]: Unsure about the naming of this

File: src/assets/js/main.js [hunk 2]
```diff
@@ -86,9 +86,43 @@
...
+    }
+
+    function historicCallback() {
+        return callbackGenerator(onHistoricDataLoaded);
+    }
+
+    d3.select('#type-selection')
+        .on('change', function() {
+            var type = d3.select(this).property('value');
+            if (type === 'bitcoin') {
```
Comment (janakerman) [inline src/assets/js/main.js]: Tidy up as discussed.

File: src/assets/js/data/feed/coinbase/historicFeed.js (hunk 1 above)
Comment (janakerman) [inline src/assets/js/data/feed/coinbase/historicFeed.js]: callbackInvalidator? Doesn't seem critical either way

File: src/assets/js/main.js (hunk 2 above)
Comment (CBarrett2) [inline src/assets/js/main.js]: done :)

**Comment by CBarrett2** [comment]
Resolves #83

**Comment by janakerman** [comment]
Looks like you've forgotten to delete `historicFeed.js`?

**Comment by janakerman** [comment]
I think a real improvement to this would be to add promises. Using promises would tidy up a lot of the async behaviour here and **may** even give you the ability to cancel callbacks. Hoever, as we discussed, it's fine to leave this for the future if we find ourselves adding more async behaviour.

**Comment by CBarrett2** [comment]
Okay :) I'll have a look into them

**Pull request merged**